oriedita/src/main/java/oriedita/editor/handler/MouseHandlerAxiom5.java

```diff
@@ -0,0 +1,246 @@
+                    OritaCalc.determineLineSegmentDistance(p, d.getLineStep().get(4)) < d.getSelectionDistance()) {
+                LineSegment s = d.get_moyori_step_lineSegment(p, 4, 5);
+                s.set(s.getB(), s.getA(), d.getLineColor());
+                s.set(OritaCalc.fullExtendUntilHit(d.getFoldLineSet(), s));
+
+                d.addLineSegment(s);
+                d.record();
+                d.getLineStep().clear();
+            }
+        }
+
+        // Case 2: Click on destination line and check of mouse point is closer to one of the 2 purple indicators
+        if(d.getLineStep().size() == 6){
+            Point intersectPoint1 = new Point(OritaCalc.findIntersection(d.getLineStep().get(3), d.getLineStep().get(5)));
+            Point intersectPoint2 = new Point(OritaCalc.findIntersection(d.getLineStep().get(4), d.getLineStep().get(5)));
+
+            double d1 = OritaCalc.distance(p, intersectPoint1);
+            double d2 = OritaCalc.distance(p, intersectPoint2);
+            d.addLineSegment(new LineSegment(d.getLineStep().get(2).getA(), d1 < d2 ? intersectPoint1 : intersectPoint2, d.getLineColor()));
+            d.record();
+            d.getLineStep().clear();
+        }
+    }
+    public void drawAxiom5FoldIndicators(double radius, Point target, LineSegment targetSegment, Point pivot) {
+        // Make sure circle radius is not 0
+        if(radius > Epsilon.UNKNOWN_1EN7){
+            LineSegment secondTemp = new LineSegment(pivot, targetSegment.determineClosestEndpoint(pivot)); //  for checks alignment between pivot point and target segment
+            double length_a = 0.0; //Distance between the center of the circle and target segment
+            Point center = new Point(pivot);
+
+            // If pivot point is not within the target segment span
+            if(OritaCalc.isLineSegmentParallel(secondTemp, targetSegment) != OritaCalc.ParallelJudgement.PARALLEL_EQUAL){
+                length_a = OritaCalc.distance(center, OritaCalc.findProjection(targetSegment, center));
+            }
+
+            // Intersect at one point
+            if(Math.abs(length_a - radius) < Epsilon.UNKNOWN_1EN7){
+                Point projectionPoint = OritaCalc.findProjection(targetSegment, pivot);
+                LineSegment projectionLine = new LineSegment(pivot, projectionPoint);
+
+                LineSegment s = new LineSegment();
+
+                if(OritaCalc.isLineSegmentParallel(new LineSegment(pivot, target), projectionLine) == OritaCalc.ParallelJudgement.NOT_PARALLEL){
+                    s.set(OritaCalc.fullExtendUntilHit(d.getFoldLineSet(), new LineSegment(pivot, OritaCalc.center(pivot, target, projectionPoint), d.getLineColor())));
+                } else{
+                    s.set(OritaCalc.fullExtendUntilHit(d.getFoldLineSet(), new LineSegment(pivot, projectionPoint, d.getLineColor())));
+                }
+
+                d.addLineSegment(s);
```

Comment:
the name of the method suggests that it only draws a preview, but it also adds lines to the actual cp here. is it possible to move this back into mouseReleased somehow?

Reply:
I'd somehow have to return the LineSegment and use it outside, which is honestly kinda painful to deal with now. Another way is to just do `d.addLineStep()` like usual and it would fit the method's purpose.